runhouse/servers/env_servlet.py

```diff
@@ -19,18 +20,53 @@
 from runhouse.resources.resource import Resource
 from runhouse.rns.utils.api import ResourceVisibility
 
-from runhouse.rns.utils.names import _generate_default_name
+# from runhouse.rns.utils.names import _generate_default_name
 from runhouse.servers.http.http_utils import (
     b64_unpickle,
+    deserialize_data,
     Message,
     OutputType,
     pickle_b64,
     Response,
+    serialize_data,
 )
 
 logger = logging.getLogger(__name__)
 
 
+def error_handling_decorator(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        EnvServlet.register_activity()
+        serialization = kwargs.pop("serialization", None)
+        serialized_data = kwargs.pop("data", None)
+        deserialized_data = deserialize_data(serialized_data, serialization)
+        kwargs["data"] = deserialized_data
+
+        try:
+            output = func(*args, **kwargs)
+            if output is not None:
+                serialized_data = serialize_data(output, serialization)
+                return Response(
+                    output_type=OutputType.RESULT_SERIALIZED,
+                    data=serialized_data,
+                    serialization=serialization,
+                )
+            else:
+                return Response(
+                    output_type=OutputType.SUCCESS,
+                )
+        except Exception as e:
+            logger.exception(e)
+            return Response(
+                error=pickle_b64(e),
+                traceback=pickle_b64(traceback.format_exc()),
```

Comment:
Shouldn't this also use the serialization option passed above?

Reply:
currently we deserialize all exceptions the same as using pickle, client side